src/Zonotope.jl

```diff
@@ -625,15 +625,17 @@ Reachable Sets of Hybrid Systems Using a Combination of Zonotopes and Polytopes.
 The one-dimensional case is not covered by that algorithm; we manually handle
 this case, assuming that there is only one generator.
 """
-function constraints_list(Z::Zonotope{N}) where {N<:AbstractFloat}
+function constraints_list(Z::Zonotope{N}; check_full_rank::Bool=true
+                         ) where {N<:AbstractFloat}
+    G = genmat(Z)
     p = ngens(Z)
     n = dim(Z)
-    if p < n
+
+    # use fallback implementation if order < 1 or matrix is not full rank
+    if p < n || (check_full_rank && rank(G) < size(G, 2))
```

Comment:
```suggestion
    if p < n || (check_full_rank && rank(G) < n)
```
The condition for full rank is `rank(G) == min(n, p)`. In the second clause, we can assume that `p >= n` so `G` is not full rank whenever `rank(G)` is smaller than `n`.